src/detext/train/train.py

```diff
@@ -298,6 +303,17 @@ def model_fn(features, labels, mode, params):
 
     group_size_field = features['group_size'] if mode != tf.estimator.ModeKeys.PREDICT else None
 
+    # For multitask training
+    task_id_field = features.get('task_id', None)
+
+    # Update the weight with each task's weight such that weight per document = weight * task_weight
+    if params.task_id_weight_map is not None:
+        weight_adj = tf.zeros(shape=tf.shape(weight), dtype="float32")
+        for task_id, task_weight in params.task_id_weight_map.items():
+            task_mask = tf.cast(tf.equal(task_id_field, task_id), dtype=tf.float32)
+            weight_adj += task_weight * task_mask
```

Comment:
This `for ... ` loop makes me a bit confused and what does adj mean here? adjusted? Just to confirm, the task_weight * task_mask will be non zero once in the loop right?

In fact you can use a implementation such as
task_weights = params.task_weights
task_ids = params.task_ids
task_mask = tf.cast(tf.equal(task_weight, task_id)...
weight_adj = tf.reduce_sum(task_mask*task_weight)

This is easier to read and more of a vectorized programming style

Reply:
To avoid the loop, I will use tf.broadcast to expand task_id_field into a shape of [batch_size, num_tasks] and then perform similar ops and tf.reduce_sum. Just want to clarify that both task_weight and task_id are lists, e.g. [0, 1, 2] for task_id, [0.1, 0.3, 0.6] for task_weight, where task_id_field is a tensor of shape [batch_size, ].